### ISSUE TITLE
"Open Instance folder" button does not work

### ISSUE THREAD

**Issue by pd2s** [PrismLauncher/PrismLauncher#1490]
### Operating System

Linux

### Version of Prism Launcher

7.2

### Version of Qt

6.5.2

### Description of bug

"Open Instance folder" button does not work on Arch with Hyprland, pcmanfm. In other apps like Telegram, Firefox buttons with similar functionality works as expected. 
This does not solved issue https://wiki.archlinux.org/title/File_manager_functionality#Directories_are_not_opened_in_the_file_manager

### Steps to reproduce

press "Open Instance folder" button

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Jamalam360** [comment]
I also have the same issue on Fedora 38 (Gnome), Prism 7.2, and I use Nautilus.

Not sure on if you can change Prism's logging level, since currently it just shows:

```
     8.159 D | Opening directory "/home/james/.local/share/PrismLauncher/instances/Jamalam's Modpack"
```

**Comment by ghost** [comment]
I also have this issue with Fedora 38 (KDE), Prism 7.2, and I use Dolphin. The in game open resource pack folder also does not work.

**Comment by Jay-716** [comment]
**EDIT**: Latest git version seems to work. The file manager is successfully invoked.

Have this issue too with Arch Linux, KDE Plasma 5.27.8, Wayland, with dolphin as the default file manager.
Sorry for my poor English in the beggining. ;)

I add some log prints in the source code to debug. And I found these in my environment. 

1. This function is implemented by [`openDirectory`](https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/ui/MainWindow.cpp#L1181C25-L1181C25) function, which then call [`openDirectory`](https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/DesktopServices.cpp#L99). And the latter would call [`openUrl`](https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/DesktopServices.cpp#L164) to open the directory(as file:/// url).
2. The [`openUrl`](https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/DesktopServices.cpp#L164) function in `launcher/DesktopServices.cpp` would call the `IndirectOpen` function. 
3. And according to the context, this function `IndirectOpen` should only be called when prism launcher is running in a container (snap or flatpak). But I just run it without any containers. It wrongly calls the open function for containers only when run without container.
4. So I reckon [this line of condition](https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/DesktopServices.cpp#L169) may be wrong. Maybe the `!` is unneeded?
After removing the `!`, in other words, inversing the condition, my dolphin can be successfully invoked.
5. However, I think the `IndirectOpen` function should have working normally as directly open. But it doesn't. I check the environment variables of the child process the function forked, and found nothing uncommon. Finally, I refer to the Qt source code, only to find nothing because it is beyond my capabilities.

Hope somebody could continue working on it.


### **TL;DR**
 - Remove the `!` [here](https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/DesktopServices.cpp#L169) would solve the problem for non-container setup.
 - It may be a Qt bug as [`IndirectOpen`](https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/DesktopServices.cpp#L167) cannot invoke file managers in child process/container environment(uncertain).

**Comment by RokeJulianLockhart** [comment]
Works for me on cpe:/o:opensuse:tumbleweed:20230921.

**Comment by shellheim** [comment]
Anyone facing this on the flatpak version?

**Comment by Tsaitou7361** [comment]
Screen record:

https://github.com/PrismLauncher/PrismLauncher/assets/99069553/21290f9d-4911-43e4-94a5-f23b42dbee56

**Comment by Tsaitou7361** [comment]
> Anyone facing this on the flatpak version?

Yes, I am.

**Comment by Tsaitou7361** [comment]
On Kubuntu 23.10 and Prismlauncher 8.0, This problem still exists...

**Comment by develcooking** [comment]
Problem still present on Arch Linux with Hyprland(wlroots compositor)y thunar and the git version of prism (AUR/prismlauncher-git) (version 8.0.r82.g4db00897c-1)

Video demo of problem

https://github.com/PrismLauncher/PrismLauncher/assets/96253562/bcff7b65-65b3-46a4-b085-c1f3f3ad1067

**Comment by forkiesassds** [comment]
Happening with Prism Launcher 8.0, on KDE Plasma 5.90.0 (Plasma 6 Beta 1) using QT 6.6.1

**Comment by Scrumplex** [comment]
I have opened PR #1954. It would be great if people could test it

**Comment by shellheim** [comment]
@Scrumplex would love to do that. Could you give me the flatpak's link? I can't find it at https://github.com/PrismLauncher/PrismLauncher/actions/runs/7208566228

**Comment by DioEgizio** [comment]
> @Scrumplex would love to do that. Could you give me the flatpak's link? I can't find it at [PrismLauncher/PrismLauncher/actions/runs/7208566228](https://github.com/PrismLauncher/PrismLauncher/actions/runs/7208566228)

flatpaks are the "Prism Launcher-x86_64" artifacts in the CI

**Comment by develcooking** [comment]
@Scrumplex so I compiled your repo with the aur prismlauncher-git package as a base. But the problem is still present on my end. Idk if I compiled something wrong or have something misconfigured.

The video is showing how I compiled prism and how it wouldn't work


https://github.com/PrismLauncher/PrismLauncher/assets/96253562/a759d4e2-ac54-494a-856e-15056476799e

**Comment by Scrumplex** [comment]
> @Scrumplex so I compiled your repo with the aur prismlauncher-git package as a base. But the problem is still present on my end. Idk if I compiled something wrong or have something misconfigured.
> 
> The video is showing how I compiled prism and how it wouldn't work

Looking at the video, you compiled it using my fork's main branch.

Try replacing the source with this url instead: `git+https://github.com/Scrumplex/PrismLauncher.git#branch=fix/open-paths-immediately`

**Comment by shellheim** [comment]
The problem still persists using flatpak on Fedora 39 : 

[Screencast_20231217_141239.webm](https://github.com/PrismLauncher/PrismLauncher/assets/116480892/1fcc2a6b-5dca-44b4-8ae2-e8198d1f3ba5)  

logs: 

```
     9.614 D | Opening path QFileInfo(/home/shell/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/1.20.2)
     9.614 D | Opening URL "file:///home/shell/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances"
    11.401 D | Opening path QFileInfo(/home/shell/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/1.20.2)
    11.401 D | Opening URL "file:///home/shell/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances"
    13.239 D | Opening path QFileInfo(/home/shell/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/1.20.2)
    13.239 D | Opening URL "file:///home/shell/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances"
```

Thanks @DioEgizio for pointing me to the flatpak.

**Comment by Scrumplex** [comment]
@shellheim Can you check if other Flatpak apps can open files/directories?

**Comment by shellheim** [comment]
yeah they work just as expected.

**Comment by Scrumplex** [comment]
> yeah they work just as expected.

Can you check if you are running the right version?

You could do that by checking the build date at Help -> About Prism Launcher. If it's 2023-12-16 it's the pull request linked to this issue

**Comment by shellheim** [comment]
Ah, no I am on 2023-12-14. I used the artifact from https://github.com/PrismLauncher/PrismLauncher/actions/runs/7208566228

**Comment by Scrumplex** [comment]
> Ah, no I am on 2023-12-14. I used the artifact from [PrismLauncher/PrismLauncher/actions/runs/7208566228](https://github.com/PrismLauncher/PrismLauncher/actions/runs/7208566228)

Try https://github.com/PrismLauncher/PrismLauncher/actions/runs/7231819672

**Comment by shellheim** [comment]
Unfortunately, the problem still persists on `2023-12-16`. (with the same logs)

**Comment by develcooking** [comment]
@Scrumplex It works now. I think you could merge the branch

Greate work 👍

https://github.com/PrismLauncher/PrismLauncher/assets/96253562/69b79d2a-4a9b-4c86-85e3-ba1dbac8e490

**Comment by Fiirgen** [comment]
Now it only opens the parent of the intended folder. For example, trying to open `~/.local/share/PrismLauncher/instances` opens `~/.local/share/PrismLauncher`. That can also be seen in the video.
